src/group_theory/perm/fin.lean

```diff
@@ -226,6 +248,32 @@ i.cycle_range.injective (by simp)
   i.cycle_range.symm j.succ = i.succ_above j :=
 i.cycle_range.injective (by simp)
 
+lemma is_cycle_cycle_range {n : ℕ} {i : fin (n + 1)} (h0 : i ≠ 0) : is_cycle (cycle_range i) :=
+begin
+  cases i with i hi,
+  cases i,
+  { exact (h0 rfl).elim },
+  exact is_cycle_fin_rotate.extend_domain _,
+end
+
+@[simp] lemma cycle_type_cycle_range {n : ℕ} {i : fin (n + 1)} (h0 : i ≠ 0) :
+  cycle_type (cycle_range i) = {i + 1} :=
+begin
+  cases i with i hi,
+  cases i,
+  { exact (h0 rfl).elim },
+  rw [cycle_range, cycle_type_extend_domain],
+  exact cycle_type_fin_rotate,
+end
+
+lemma is_three_cycle_cycle_range_two {n : ℕ} :
+  is_three_cycle (cycle_range 2 : perm (fin (n + 3))) :=
```

Comment:
Would it make sense to have a hypothesis on n, rather than use n+3?

Reply:
I need to use at least `n+2` to use the numeral 2.